tests/test_samples.py

```diff
@@ -526,20 +526,31 @@ def test_plot_logscale_2d(kind):
 def test_logscale_ticks():
     np.random.seed(42)
     ndim = 5
-    data = np.exp(10 * np.random.randn(200, ndim))
+    data1 = np.exp(10 * np.random.randn(200, ndim))
+    data2 = np.exp(10 * np.random.randn(200, ndim) - 50)
     params = [f'a{i}' for i in range(ndim)]
     fig, axes = make_2d_axes(params, logx=params, logy=params, upper=False)
-    samples = Samples(data, columns=params)
-    samples.plot_2d(axes)
-    for _, col in axes.iterrows():
-        for _, ax in col.items():
+    samples1 = Samples(data1, columns=params)
+    samples2 = Samples(data2, columns=params)
+    samples1.plot_2d(axes)
+    samples2.plot_2d(axes)
+    for y, col in axes.iterrows():
+        for x, ax in col.items():
             if ax is not None:
                 xlims = ax.get_xlim()
                 xticks = ax.get_xticks()
                 assert np.sum((xticks > xlims[0]) & (xticks < xlims[1])) > 1
                 ylims = ax.get_ylim()
                 yticks = ax.get_yticks()
                 assert np.sum((yticks > ylims[0]) & (yticks < ylims[1])) > 1
+                if x == y:
+                    data_min = ax.twin.dataLim.intervalx[0]
+                    data_max = ax.twin.dataLim.intervalx[1]
+                    assert xlims[0] == pytest.approx(data_min, rel=1e-14)
+                    assert xlims[1] == pytest.approx(data_max, rel=1e-14)
+                else:
+                    assert_array_equal(xlims, ax.dataLim.intervalx)
+                    assert_array_equal(ylims, ax.dataLim.intervaly)
```

Comment:
CI initially fails after adding this test, see CI of https://github.com/handley-lab/anesthetic/pull/383/commits/5585530fae43ddeb6789dceace5cbf811dc1e870